spec/server/listener_metrics_spec.rb

```diff
@@ -162,6 +168,14 @@ def stop_listener(port, server)
         sleep(2) # Give it some time to stop
       end
 
+      def wait_for_puma_control_socket
+        Timeout::timeout(5) do
+          until system("bundle exec bin/3scale_backend status")
```

Comment:
Improvement: wait until the control socket can be opened and is writable.

Reply:
That's basically what the status command does. It returns an error when the control socket cannot be opened.